src/security.ts

```diff
@@ -167,3 +170,60 @@ export const iamMfaEnabledForConsoleAccess: ResourceValidationPolicy = {
     }),
 };
 registerPolicy("iamMfaEnabledForConsoleAccess", iamMfaEnabledForConsoleAccess);
+
+
+/** @internal */
+// Enforce the note on aws.iam.Role:
+//
+// NOTE: If you use this resource’s managed_policy_arns argument or inline_policy configuration blocks, this resource
+// will take over exclusive management of the role's respective policy types (e.g., both policy types if both arguments
+// are used). These arguments are incompatible with other ways of managing a role's policies, such as
+// aws.iam.PolicyAttachment, aws.iam.RolePolicyAttachment, and aws.iam.RolePolicy. If you attempt to manage a role’s
+// policies by multiple means, you will get resource cycling and/or errors.
+export const iamRoleNoPolicyManagementConflicts: StackValidationPolicy = {
```

Comment:
Probably does not belong in security.ts? Should it live somewhere else?

Reply:
maybe

- `warnings.ts`
- `configuration.ts`
- `config-warning.ts`

Reply:
or something IAM specific? I see that the repo has files per service/domain (e.g. elasticsearch.ts, compute.ts)